storyblocks/object_detection/model/code/inference.py

```diff
@@ -0,0 +1,294 @@
+    PARAMS = yaml.safe_load(fp)
+
+print(f'PARAMS = {PARAMS}')
+
+# input size is fixed on the compound coef level
+MAX_INPUT_SIZES = [512, 640, 768, 896, 1024, 1280, 1280, 1536, 1536]
+MAX_INPUT_SIZE = MAX_INPUT_SIZES[EFFICIENTDET_COMPOUND_COEF]
+print(f"MAX_INPUT_SIZE = {MAX_INPUT_SIZE}")
+
+
+def log_storyblocks_artifact_info():
+    """if a face detection repo sha or archive timestamp file exist, print
+    their contents"""
+    print('looking for archive artifact metadata')
+    here = os.path.dirname(os.path.realpath(__file__))
+
+    for fname in ['model_archive_timestamp', 'yaefd_current_sha']:
+        fname_full = os.path.join(here, fname)
+        if os.path.isfile(fname_full):
+            with open(fname_full, 'r') as fp:
+                print(f"{fname}: {fp.read().strip()}")
+
+
+log_storyblocks_artifact_info()
+
+
+class StoryblocksCustomError(errors.GenericInferenceToolkitError):
+    def __init__(self, message):
+        super().__init__(400, message)
+
+
+def get_weights_url(c):
+    """this list of urls is hand-created from
+    https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/blob/master/readme.md as of the
+    time of writing this file (2020-12-03). as the model is updated, these urls *could* (but
+    likely won't) change much. of course we may need to update this over time
+
+    this could be automated with the github releases api, but that's overkill rn
+    https://developer.github.com/v3/repos/releases/
+
+    Args
+        c: the compound coefficient (read the efficientnet/det papers for details)
+
+    Returns
+        url of the weights file to download
+
+    """
+    url_map = {
+        '0': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
+             '/efficientdet-d0.pth',
+        '1': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
+             '/efficientdet-d1.pth',
+        '2': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
+             '/efficientdet-d2.pth',
+        '3': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
+             '/efficientdet-d3.pth',
+        '4': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
+             '/efficientdet-d4.pth',
+        '5': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
+             '/efficientdet-d5.pth',
+        '6': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
+             '/efficientdet-d6.pth',
+        '7': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.2'
+             '/efficientdet-d7.pth',
+        '8': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.2'
+             '/efficientdet-d8.pth',
+        '7x': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.2'
```

Comment:
What's the difference between 8 and 7x?

Reply:
none -- the "real" name is `7x` and the library just uses `8` as a shorthand so that the numeric indexing still works.

will bite them in the ass when `d8` actually comes out, lol